jgrapht-core/src/test/java/org/jgrapht/util/AsSynchronizedIterateOnCopySetTest.java

```diff
@@ -0,0 +1,142 @@
+package org.jgrapht.util;
+
+import java.util.*;
+import org.junit.*;
+import static org.junit.Assert.*;
+
+/**
+ * Test class AsSynchronizedIterateOnCopySet.
+ *
+ * @author CHEN Kui
+ */
+public class AsSynchronizedIterateOnCopySetTest
+{
+    @Test
+    public void testAdd()
+    {
+        AsSynchronizedIterateOnCopySet<Integer> set = new AsSynchronizedIterateOnCopySet<>(new HashSet<>());
+        class Add
+            implements Runnable
+        {
+            private int start, end;
+            private Add(int _start, int _end)
+            {
+                start = _start;
+                end = _end;
+            }
+            @Override
+            public void run()
+            {
+                for (int i = start; i < end; i++) {
+                    try {
+                        Thread.sleep(50);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                    set.add(i);
+                }
+            }
+        }
+
+        for (int i = 0; i < 20; i++) {
+            new Thread(new Add(20*i, 20*i + 20)).start();
+        }
+        while (Thread.activeCount() > 2);
```

Comment:
This is not a correct way to wait for threads to complete.  Take a look at the Thread.join method, but you should instead be using a real multi-threaded JUnit test framework (such as ActiveTestSuite).